backend/plonk/bn254/solidity.go

```diff
@@ -236,9 +317,10 @@ contract PlonkVerifier {
       // The data described above is written starting at mPtr. "gamma" lies on 5 bytes,
       // and is encoded as a uint256 number n. In basis b = 256, the number looks like this
       // [0 0 0 .. 0x67 0x61 0x6d, 0x6d, 0x61]. The first non zero entry is at position 27=0x1b
-      function derive_gamma(aproof, pub_inputs) {
+      function derive_gamma(aproof, ins)->gamma_not_reduced {
         
-        let mPtr := mload(0x40)
+        let state := mload(0x40)
+        let mPtr := add(mload(0x40), state_last_mem)
```

Comment:
Couldn't we use `state` as the first argument here?

Reply:
yep 👍